sdk/core/azure-json/src/main/java/com/azure/json/JsonObject.java

```diff
@@ -0,0 +1,494 @@
+//            }
+//            // Case: element is not null, but is also not a valid type that can
+//            // be cast or converted to a valid JsonElement type.
+//            else { throw new IllegalArgumentException(); }
+//        }
+//        // Case: element is null, therefore value is a JsonNull object.
+//        else { value = new JsonNull(); }
+//
+//        // Adding the new property, the key with its value pair. The value is
+//        // the respective JsonElement cast/conversion of element.
+//        this.addProperty(key, value);
+//        //properties.put(key, value);
+//
+//        // Returning the new state of the JsonObject after the successful
+//        // addition of the new property
+//        return this;
+//    }
+
+//    /**
+//     * Adds a new property into the JsonObject object using JsonElement as the
+//     * property value.
+//     *
+//     * The primary user is the build method of JsonObject and JsonArray.
+//     *
+//     * @param key
+//     * @param element
+//     * @return The new state of the JsonObject after the successful addition of
+//     * the new property
+//     * @throws IllegalArgumentException
+//     */
+//    public JsonObject addProperty(String key, JsonElement element) throws IllegalArgumentException {
+//        // Adding the new property, the key with its value pair. The value is
+//        // the respective JsonElement cast/conversion of element.
+//        properties.put(key, element);
+//        return this;
+//    }
+
+    /**
+     * Returns a property value for the corresponding specified property key
+     * from the JsonObject.
+     *
+     * @param key Specifies the property key that identifies the property in the
+     * JsonObject.
+     * @return JsonElement representing the value of the property specified by
+     * the key.
+     *
+     * TODO: this method currently does what the new getValueBykey method does.
+     * This getProperty method should do what it suggests, which is get a property.
+     * Properties are the whole key-value pair not just their key or value.
+     * Therefore, this method needs to be redefined to return key value pairs,
+     * for example Entry objects since properties is currently utilising HashMap.
+     * Perhaps we could confirm with the dev team on discord if this method is
+     * required to behave in the current way that it is, if so, then we could
+     * keep this method as it is and remove the getKeyByValue and getValueByKey
+     * methods that I have defined.
+     */
+    public JsonElement getProperty(String key) {
+        return properties.get(key);
+    }
+
+    /**
+     * Gets the value for a given property by key.
+     *
+     * @param key specifying the key to match a property with to find the
+     * respective value pair of
+     * @return JsonElement object representing the respective value pair of the
+     * search key
+     */
+    public JsonElement getValueByKey(String key) {
```

Comment:
What's the difference between `getValueByKey` and `getProperty`?

Reply:
@srnagar they are identical, getValueByKey should be removed. At the start of the project I created two methods in JsonObject called getValueByKey and getKeyByValue. I created them because I thought it would be better to be more accurate in the naming because a property in JSON refers to a key and value pair in combination and not just one or the other. So, there was two methods, one to get a property's key and one to get the property's value. However, it was settled that we were just going to make "property" synonymous with a property's value despite being less accurate in JSON terminology. getKeyByValue was removed a while ago because it was not deemed useful, and getValueByKey should also have been removed because of the introduction of getProperty as a replacement.